src/python/pants/backend/python/macros/poetry_requirements_test.py

```diff
@@ -0,0 +1,414 @@
+    poetry = {git = "https://github.com/python-poetry/poetry.git", tag = "v1.1.1"}
+    requests = {extras = ["security"], version = "^2.25.1", python = ">2.7"}
+    foo = [{version = ">=1.9", python = "^2.7"},{version = "^2.0", python = "3.4 || 3.5"}]
+
+    [tool.poetry.dependencies.black]
+    version = "19.10b0"
+    python = "3.6"
+    markers = "platform_python_implementation == 'CPython'"
+
+    [tool.poetry.dev-dependencies]
+    isort = ">=5.5.1,<5.6"
+
+    [build-system]
+    requires = ["poetry-core>=1.0.0"]
+    build-backend = "poetry.core.masonry.api"
+    """
+    retval = parse_pyproject_toml(toml_str, "/path/to/file")
+    actual_reqs = {
+        Requirement.parse("python<4.0.0,>=3.8"),
+        Requirement.parse("junk@ https://github.com/myrepo/junk.whl"),
+        Requirement.parse("poetry@ git+https://github.com/python-poetry/poetry.git@v1.1.1"),
+        Requirement.parse('requests<3.0.0,>=2.25.1; python_version > "2.7"'),
+        Requirement.parse('foo>=1.9; python_version >= "2.7" and python_version < "3.0"'),
+        Requirement.parse('foo<3.0.0,>=2.0; python_version == "3.4" or python_version == "3.5"'),
+        Requirement.parse(
+            'black==19.10b0; platform_python_implementation == "CPython" and python_version == "3.6"'
+        ),
+        Requirement.parse("isort<5.6,>=5.5.1"),
+    }
+    assert retval == actual_reqs
+
+
+@pytest.fixture
+def rule_runner() -> RuleRunner:
+    return RuleRunner(
+        rules=[QueryRule(Targets, (Specs,))],
+        target_types=[PythonRequirementLibrary, PythonRequirementsFile],
+        context_aware_object_factories={"poetry_requirements": PoetryRequirements},
+    )
+
+
+def assert_poetry_requirements(
+    rule_runner: RuleRunner,
+    build_file_entry: str,
+    pyproject_toml: str,
+    *,
+    expected_file_dep: PythonRequirementsFile,
+    expected_targets: Iterable[PythonRequirementLibrary],
+    pyproject_toml_relpath: str = "pyproject.toml",
+) -> None:
+    rule_runner.add_to_build_file("", f"{build_file_entry}\n")
+    rule_runner.create_file(pyproject_toml_relpath, pyproject_toml)
+    targets = rule_runner.request(
+        Targets,
+        [Specs(AddressSpecs([DescendantAddresses("")]), FilesystemSpecs([]))],
+    )
+    assert {expected_file_dep, *expected_targets} == set(targets)
+
+
+def test_pyproject_toml(rule_runner: RuleRunner) -> None:
+    """This tests that we correctly create a new python_requirement_library for each entry in a
+    pyproject.toml file.
+
+    Note that this just ensures proper targets are created; see prior tests for specific parsing
+    edge cases.
+    """
+    assert_poetry_requirements(
+        rule_runner,
+        "poetry_requirements(module_mapping={'ansicolors': ['colors']})",
+        dedent(
+            """\
+            [tool.poetry.dependencies]
+            Django = {version = "3.2", python = "3"}
+            ansicolors = ">=1.18.0"
+            Un-Normalized-PROJECT = "1.0.0"
```

Comment:
You could move one of these into `dev-dependencies` to be really sure we handle those too. You do already have a unit test for that, but it's a pretty important requirement we handle it correctly so doesn't hurt to test again here in this integration test.